ibis_substrait/compiler/translate.py

```diff
@@ -777,13 +777,32 @@ def selection(
 
     if selections := _get_selections(op):
 
-        if relation.project.common.ListFields():
-            # if there is already an `emit` in RelCommon then we're stacking
-            # projections and we need to update the output_mapping to refer to
-            # the number of fields present in the most recent emit
-            mapping_counter = itertools.count(
-                len(relation.project.common.emit.output_mapping)
+        rels = [
```

Comment:
Why do need to have a list of predefined node types instead of doing it for all the node types? (I thought emit output_mapping is a thing that should work for all node types?)

Reply:
it was just simpler than querying all of the attributes of the object (also because google does weird things in protobuf and creates attributes that don't exist)